pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -279,6 +275,24 @@ func (ctrl *VMSnapshotController) updateVMSnapshotContent(content *snapshotv1.Vi
 
 	if ready && (contentCpy.Status.ReadyToUse == nil || !*contentCpy.Status.ReadyToUse) {
 		contentCpy.Status.CreationTime = currentTime()
+
+		source, err := ctrl.getSnapshotSource(vmSnapshot)
+		if err != nil {
+			return 0, err
+		}
+
+		frozen, err := source.Frozen()
```

Comment:
This code will only happen once, if the vm status havnt changed yet to frozen (which can take up to 5 secs) then this will be false and the unfrozen will never occure

Reply:
maybe if online and guest agent you can assume its frozen at this point and just unfreeze it wont fail if it is already unfrozen..

Reply:
Ah right since we are not waiting to observe the VMI being frozen we may miss the unfreeze here hmmmm